subsys/net/ip/net_if.c

```diff
@@ -2890,9 +2890,16 @@ const struct in6_addr *net_if_ipv6_select_src_addr(struct net_if *dst_iface,
 		if (dst_iface) {
 			src = net_if_ipv6_get_ll(dst_iface, NET_ADDR_PREFERRED);
 		} else {
-			STRUCT_SECTION_FOREACH(net_if, iface) {
-				struct in6_addr *addr;
+			struct in6_addr *addr;
+
+			addr = net_if_ipv6_get_ll(net_if_get_default(), NET_ADDR_PREFERRED);
 
```

Comment:
This empty line can be removed

Reply:
Done

Reply:
Hmm, I do not see the line removed. I would like to see something like this here:
```
addr = ....;
if (addr) {
```
We try to keep the variable setting and then checking it "together", so there is no need to place empty line between them.